playground/backend/cmd/server/controller.go

```diff
@@ -87,19 +87,48 @@ func (controller *playgroundController) CheckStatus(ctx context.Context, info *p
 	return &pb.CheckStatusResponse{Status: status}, nil
 }
 
-//GetRunOutput is returning output of execution for specific pipeline by PipelineUuid
+// GetRunOutput is returning output of execution for specific pipeline by PipelineUuid
 func (controller *playgroundController) GetRunOutput(ctx context.Context, info *pb.GetRunOutputRequest) (*pb.GetRunOutputResponse, error) {
 	pipelineId := info.PipelineUuid
+	runOutputLastIndex, err := controller.cacheService.GetValue(ctx, uuid.MustParse(pipelineId), cache.RunOutputIndex)
+	if err != nil {
+		logger.Errorf("%s: GetRunOutput(): cache.GetValue: error: %s", pipelineId, err.Error())
+		return nil, errors.NotFoundError("GetRunOutput", "there is no run output for pipelineId: "+pipelineId+", subKey: cache.RunOutputLastIndex")
+	}
```

Comment:
What happens if it is the first time we try to retrieve the `runOutputLastIndex`? Does it become equal to 0?

Reply:
If we are trying to receive some value from cache (using key and subKey), but cache doesn't contain the combination of (key:subkey) then we will receive an error (example from `local_cache`):
```
value, found := lc.items[pipelineId][subKey]
	if !found {
		lc.RUnlock()
		return nil, fmt.Errorf("value with pipelineId: %s and subKey: %s not found", pipelineId, subKey)
	}
```
Does it answer your question?

Reply:
I got it – I mean, if we have such a combination of (key:subKey) in cache but we never retrieved it before, what happens with `runOutputLastIndex`?

Reply:
We put `0` to cache as a value for `{pipelineId}:cache.RunOutputIndex` combination in `RunCode()` method. So if we call `GetRunOutput()` after `RunCode()` then cache should contain value for `{pipelineId}:cache.RunOutputIndex`.